src/routers/alpha-router/alpha-router.ts

```diff
@@ -1001,9 +1001,17 @@ export class AlphaRouter
       const quoteDiff = swapRouteFromChain.quote.subtract(swapRouteFromCache.quote);
       const quoteGasAdjustedDiff = swapRouteFromChain.quoteGasAdjusted.subtract(swapRouteFromCache.quoteGasAdjusted);
       const gasUsedDiff = swapRouteFromChain.estimatedGasUsed.sub(swapRouteFromCache.estimatedGasUsed);
+      const gasAdjustedDiffInAmountTokenTerms = quoteGasAdjustedDiff.multiply(amount.divide(swapRouteFromChain.quoteGasAdjusted));
+      const misquotePercent = gasAdjustedDiffInAmountTokenTerms.divide(amount).multiply(100);
```

Comment:
hard for me to follow whether this is JSBI or BigNumber being used, but I know `BigNumber` doesn't have a concept of decimals, so if these are intended to go below 0, then back up, I think the number would get lost??

Reply:
Ok I see they're CurrencyAmount/Fraction. I feel like there's all kinds of things going on in SOR it can be dizzying (since always haha), I wonder if in general it might be helpful to define types more strictly just for readability.

Just a general thought no need to address now

Reply:
I personally, as not the biggest math wiz ever, find the calculations and reasoning for them hard to follow, like why must we get misquotePercent with "amountToken" terms.

thinking more strict types and maybe some comments might be helpful

Reply:
thanks for all the feedback, you are right the misquotePercent is a currencyAmount and that's wonky. I'm mostly just trying to do some math here.

What I'm doing is getting the diff between the cache and regular quote, then converting that to the amountIn terms, e.g. if a user is sending 1 ETH, then we get the quote in both methods, we calculate the difference of that quote (which is in tokenOut terms), but to be able to compare against ETH I'm then multiplying the difference by the price of tokenOut with respect to tokenIn (amount/quote).

With that being said, it appears like I can simplify this math.

But essentially what I'm interested in getting is the percent of misquote calculated by the difference and what it would have received in reality